services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler_utils.py

```diff
@@ -100,7 +99,29 @@ async def discover_running_services(schduler: "Scheduler") -> None:  # type: ign
     logger.info("The following services need to be observed: %s", services_to_observe)
 
     for scheduler_data in services_to_observe:
-        await schduler._add_service(scheduler_data)  # pylint: disable=protected-access
+        await scheduler._add_service(scheduler_data)  # pylint: disable=protected-access
+
+
+def create_model_from_scheduler_data(
+    node_uuid: NodeID,
+    scheduler_data: SchedulerData,
+    service_state: ServiceState,
+    service_message: str,
+) -> RunningDynamicServiceDetails:
+    return RunningDynamicServiceDetails.parse_obj(
+        {
+            "boot_type": ServiceBootType.V2,
+            "user_id": scheduler_data.user_id,
+            "project_id": scheduler_data.project_id,
+            "service_uuid": node_uuid,
+            "service_key": scheduler_data.key,
+            "service_version": scheduler_data.version,
+            "service_host": scheduler_data.service_name,
+            "service_port": scheduler_data.service_port,
+            "service_state": service_state.value,
+            "service_message": service_message,
+        }
+    )
```

Comment:
I really don't see the point in moving this. Nor do I see the point in moving `RunningDynamicServiceDetails` to the models library. Since this one is only used by the `director-v2`.

Reply:
Ok, perhaps I got thisone wrong. I intend to move ONLY models that are needed in the schema. `RunningDynamicServiceDetails` is not part of the schema?

Reply:
These are the models used in the routes `/dynamic_services`:
https://github.com/ITISFoundation/osparc-simcore/blob/383a6803825d89056500f4fc6055b998951bb6f0/services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_services.py#L9-L13

According to this, ``RunningDynamicServiceDetails`` is an alias of ``DynamicServiceGet``, therefore it should be in `models_library`

https://github.com/ITISFoundation/osparc-simcore/blob/706dcfc1c67e18d116117713bbd2c643ed41f610/packages/models-library/src/models_library/api_schemas_directorv2/dynamic_services.py#L63


What I intended by moving `create_model_from_scheduler_data` as a free function was indeed to reduce the coupling since the function included a class `SchedulerData` which is an internal model (and therefore lives in the service and not in the models_library package)


Does it makes sense now?

Reply:
Sorry, wrong model. I0ve confused it with something different. You can move that one if you want.